### ISSUE TITLE
Strange behavior when starting localstack via docker-compose

### ISSUE THREAD

**Issue by DIU-marc** [localstack/localstack#5009]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We're using localstack PRO.

When I start localstack via docker-compose (using SERVICES to define needed services), Lambda and DynamoDB won't start. /health says only "available" not "running".

After update Lambda code, the Lambda Service is "running".

After visiting localstack.cloud and click on "Resources/DynamoDB", also dynamodb starts running.

```
|2021-11-25T13:06:29.174:DEBUG:localstack.utils.run:Executingcommand:cd"/opt/code/localstack/localstack/infra/dynamodb"&&zip-uDynamoDBLocal.jarlog4j2.xml||true
|2021-11-25T13:06:29.292:DEBUG:localstack.services.dynamodb.server:startingdynamodbprocess['java','-Xmx256m','-Djava.library.path=/opt/code/localstack/localstack/infra/dynamodb/DynamoDBLocal_lib','-jar','/opt/code/localstack/localstack/infra/dynamodb/DynamoDBLocal.jar','-port','44425','-dbPath','/tmp/localstack/data/dynamodb']
|2021-11-25T13:06:29.293:DEBUG:localstack.utils.run:Executingcommand:['java','-Xmx256m','-Djava.library.path=/opt/code/localstack/localstack/infra/dynamodb/DynamoDBLocal_lib','-jar','/opt/code/localstack/localstack/infra/dynamodb/DynamoDBLocal.jar','-port','44425','-dbPath','/tmp/localstack/data/dynamodb']
|2021-11-25T13:06:29.294:INFO:localstack.services.infra:StartingmockDynamoDBserviceonhttpports443/4566...
|2021-11-25T13:06:30.420:INFO:localstack.services.dynamodb.server:InitializingDynamoDBLocalwiththefollowingconfiguration:
|2021-11-25T13:06:30.421:INFO:localstack.services.dynamodb.server:Port:44425
|2021-11-25T13:06:30.421:INFO:localstack.services.dynamodb.server:InMemory:false
|2021-11-25T13:06:30.421:INFO:localstack.services.dynamodb.server:DbPath:/tmp/localstack/data/dynamodb
|2021-11-25T13:06:30.422:INFO:localstack.services.dynamodb.server:SharedDb:false
|2021-11-25T13:06:30.422:INFO:localstack.services.dynamodb.server:shouldDelayTransientStatuses:false
|2021-11-25T13:06:30.425:INFO:localstack.services.dynamodb.server:CorsParams:*
|2021-11-25T13:06:30.478:INFO:localstack.services.dynamodb.server:
```

Invoking a Lambda function which requests the dynamodb, ends in a "NetworkingError" (Timeout).

```
Error: connect ETIMEDOUT 172.17.0.1:4566
     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
   errno: -110,
   code: 'NetworkingError',
   syscall: 'connect',
   address: '172.17.0.1',
   port: 4566,
   region: 'us-east-1',
   hostname: '172.17.0.1',
   retryable: true,
   time: 2021-11-25T11:13:01.876Z
}
```

My NodeJS code is (the table contains 3 items, added via localstack.cloud):

```
let docClient = new AWS.DynamoDB.DocumentClient({
    region: "us-east-1",
    endpoint: process.env.AWS_ENDPOINT_URL
  });

  let params = {
    TableName: "foobar",
    Key: {
      "id": "120"
    }
  };

  docClient.get(params, function(err, data) {
    if (err) {
      console.error("Unable to read item. Error JSON:", JSON.stringify(err, null, 2));
    } else {
      console.log("GetItem succeeded:", JSON.stringify(data, null, 2));
    }
  });
```

I also tried http://${process.env.LOCALSTACK_HOSTNAME}:4566  as endpoint- same error (also without using endpoint, also with/without region). 

I also added LAMBDA_DOCKER_NETWORK.

### Expected Behavior

Running DynamoDB and Lambda Services after starting localstack.

I can't specify the problem with the Lambda/DynamoDB connection, need more debugging aout that issue - but it seems to be a network problem.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker-compose up

```
localstack:
    image: localstack/localstack:latest
    ports:
      - "4566:4566"
      - "4571:4571"
      - "4572:4572"
    environment:
      AWS_REGION: ${AWS_REGION}
      AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
      AWS_SECRET_KEY: ${AWS_SECRET_KEY}
      HOSTNAME: localstack
      SERVICES: ec2,dax,lambda,apigateway,dynamodb,dynamodbstreams,s3,sns,iam,kinesis,dynamodb
      DOCKER_HOST: unix:///var/run/docker.sock
      LAMBDA_EXECUTOR: docker-reuse
      LAMBDA_DOCKER_NETWORK: "my-network"
      DATA_DIR: /tmp/localstack/data
    volumes:
      - ./.localstack:/tmp/localstack
      - /var/run/docker.sock:/var/run/docker.sock
    container_name: localstack.jobflow-indesign-service
    networks:
      - "my-network"
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- OS: Ubuntu 20.04
- LocalStack: 0.13 (latest)
```

### Anything else?

I'm not able to install localstack package from python:
Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-FhtDCA/localstack/setup.py", line 5, in <module>
        from plugin.setuptools import load_entry_points
    ImportError: No module named plugin.setuptools

**Comment by DIU-marc** [comment]
Update:

If I run the above node snippet from my host, the connection does work:

$ node test.js 
GetItem succeeded: { Item: { name: 'foobar', id: '120' } }

But from inside a Lambda Container it does not. Can't get to debug, I can't install anything inside a lamci container.

**Comment by DIU-marc** [comment]
Update2:
using network ID instead of natwork NAME seems to work now. Only services don't start every time localstack starts.

**Comment by naspeh** [comment]
Hi @DIU-marc 

[From localstack v0.13.0](https://github.com/localstack/localstack/releases/tag/v0.13.0)
> * Lazy-loading of services as the default behavior

For previous behavior use `EAGER_SERVICE_LOADING=1` with `SERVICES=`

Thanks.

**Comment by dominikschubert** [comment]
Was fixed in #5056
